app/src/main/java/com/github/warnastrophy/core/ui/healthcard/HealthCardScreen.kt

```diff
@@ -0,0 +1,684 @@
+  const val ORGAN_DONOR_FIELD = "OrganDonorSwitch"
+  const val NOTES_FIELD = "NotesField"
+
+  // Action button tags
+  const val ADD_BUTTON = "AddButton"
+  const val UPDATE_BUTTON = "UpdateButton"
+  const val DELETE_BUTTON = "DeleteButton"
+}
+
+/**
+ * Represents the state of the Health Card form.
+ *
+ * This class manages all form fields and their validation states. It uses a "touched" pattern to
+ * track user interaction with required fields, enabling validation feedback only after the user has
+ * interacted with a field.
+ *
+ * @property fullName The full name of the card holder (required)
+ * @property fullNameTouched Whether the full name field has been interacted with
+ * @property birthDate The birth date in dd/MM/yyyy format (required)
+ * @property birthDateTouched Whether the birth date field has been interacted with
+ * @property socialSecurityNumber The social security number (required)
+ * @property ssnTouched Whether the SSN field has been interacted with
+ * @property sex The biological sex (optional)
+ * @property bloodType The blood type (optional)
+ * @property heightCm Height in centimeters (optional)
+ * @property weightKg Weight in kilograms (optional)
+ * @property chronicConditions Comma-separated list of chronic conditions (optional)
+ * @property allergies Comma-separated list of allergies (optional)
+ * @property medications Comma-separated list of medications (optional)
+ * @property onGoingTreatments Comma-separated list of ongoing treatments (optional)
+ * @property medicalHistory Comma-separated list of medical history items (optional)
+ * @property organDonor Whether the person is an organ donor
+ * @property notes Additional notes (optional)
+ */
+@Stable
+data class HealthCardFormState(
+    val fullName: String = "",
+    val fullNameTouched: Boolean = false,
+    val birthDate: String = "",
+    val birthDateTouched: Boolean = false,
+    val socialSecurityNumber: String = "",
+    val ssnTouched: Boolean = false,
+    val sex: String = "",
+    val bloodType: String = "",
+    val heightCm: String = "",
+    val weightKg: String = "",
+    val chronicConditions: String = "",
+    val allergies: String = "",
+    val medications: String = "",
+    val onGoingTreatments: String = "",
+    val medicalHistory: String = "",
+    val organDonor: Boolean = false,
+    val notes: String = ""
+) {
+  /** Regex pattern for date validation in dd/MM/yyyy format */
+  private val dateRegex = Regex("""^([0-2]\d|3[01])/(0\d|1[0-2])/(\d{4})$""")
+
+  /**
+   * Validates the birth date format and ensures it represents a valid date.
+   *
+   * @return true if the date is valid and parseable, false otherwise
+   */
+  fun isDateValid(): Boolean {
+    if (birthDate.isBlank()) return false
+    return try {
+      val formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy")
+      LocalDate.parse(birthDate, formatter)
+      true
+    } catch (e: Exception) {
+      false
+    }
+  }
```

Comment:
Here I think we should do a more robust check on `birthDate` because currently: _30/02/2000_ is considered valid. In addition, we could add a check to see if the date is **before** today's date. For example, you cannot be born on _01/01/2004_ if today is _01/01/2000_. I suggest to make the function more reusable too (so it could fit with not only birth date)

Example:  
```suggestion
private val DATE_FORMATTER = DateTimeFormatter.ofPattern("dd/MM/uuuu")
    .withResolverStyle(ResolverStyle.STRICT)
    
  fun isDateValid(input: String): Boolean {
    if (birthDate.isBlank()) return false
   try {
      val date = LocalDate.parse(input, DATE_FORMATTER)
      return !date.isAfter(LocalDate.now()) // Check if date not anterior to today
    } catch (e: Exception) {
      return false
    }
  }
```

Reply:
Done